packages/admin/admin/src/mui/menu/CollapsibleItem.tsx

```diff
@@ -1,96 +1,192 @@
 
-        return React.cloneElement<MenuLevel>(child, {
-            level: newItemLevel === 1 || newItemLevel === 2 || newItemLevel === 3 ? newItemLevel : undefined,
+    const childElements = useMemo(() => {
+        function checkIfPathInLocation(child: React.ReactElement<MenuCollapsibleItemProps | MenuItemRouterLinkProps | MenuItemProps>) {
+            return "to" in child.props && matchPath(location.pathname, { path: child.props.to, strict: true });
+        }
+        hasSelectedChild.current = false;
+
+        return React.Children.map(children, (child: MenuChild) => {
+            // child is selected
+            if (checkIfPathInLocation(child)) {
+                hasSelectedChild.current = true;
+            }
+
+            // sub child is selected
+            const subChildElements =
+                "children" in child.props ? React.Children.map(child?.props?.children, (child: MenuChild) => child) : ([] as MenuChild[]);
+            if (subChildElements?.some((child: MenuChild) => child.props && checkIfPathInLocation(child))) {
+                hasSelectedChild.current = true;
+            }
+
+            const newItemLevel = itemLevel + 1;
+
+            return React.cloneElement<MenuCollapsibleItemProps | MenuItemRouterLinkProps | MenuItemProps>(child, {
+                level: newItemLevel === 1 || newItemLevel === 2 || newItemLevel === 3 ? newItemLevel : undefined,
+                isMenuOpen,
+                isCollapsibleOpen: isSubmenuOpen,
+            });
         });
-    });
+    }, [children, isMenuOpen, isSubmenuOpen, itemLevel, location.pathname]);
+
+    const closeMenu = () => {
+        setAnchorEl(undefined);
+        setIsSubmenuOpen(false);
+    };
+
+    const handlePopoverOpen = (event: React.MouseEvent<HTMLElement>) => {
+        if (isMenuOpen) return;
+        if (anchorEl !== event.currentTarget) {
+            setAnchorEl(event.currentTarget);
+            setIsSubmenuOpen(true);
+        }
+    };
+
+    const handlePopoverClose = (e: React.MouseEvent<HTMLElement>) => {
+        if (isMenuOpen) return;
+        const el = e.target as HTMLElement;
+        const rect = el.getBoundingClientRect();
+        const { clientX, clientY } = e;
+
+        if (childElements?.length && clientX + 2 > rect.right && clientY > rect.top && clientY < rect.bottom) return;
 
-    const [open, setOpen] = React.useState<boolean>(openByDefault || hasSelectedChild);
+        closeMenu();
+    };
 
-    const listClasses = [classes.root];
-    if (hasSelectedChild) listClasses.push(classes.childSelected);
-    if (open) listClasses.push(classes.open);
+    const listClasses = [];
+    if (hasSelectedChild.current) listClasses.push(classes.childSelected);
+    if (isSubmenuOpen) listClasses.push(classes.open);
 
     return (
-        <div {...otherProps} className={listClasses.join(" ")}>
-            <div className={classes.listItem}>
+        <div {...otherProps}>
+            <div
+                id="menu-item"
```

Comment:
Ids should be unique per page, this one is being used for every item in the menu. 
Also `menu-item` is very generic, which increases the chance of colliding with another components id.